iosapp/lib-ios/Sources/LibFilter/FilterProxy.swift

```diff
@@ -77,7 +88,13 @@ public class FilterProxy {
       return .drop
     }
 
-    guard let rules = self.protectionMode.rules else {
+    // NB: this _should_ be extremely fast, because
+    //  1) network requests originate relatively rarely
+    //  2) it's likely that the filter only ever operates on a single thread
+    //  3) underlying NSLock should not incur a syscall in low contention
+    let protectionMode = self.protectionMode.withLock { $0 }
```

Comment:
this is the key part of the filter that is performance sensitive. the other locks are essentially meaningless (other than to prove to the compiler that all is safe) because they are locked an unlocked extremely rarely.